app/controllers/favorite_controller.rb

```diff
@@ -0,0 +1,34 @@
+class FavoriteController < ApplicationController
+  def update
+    pet = Pet.find(params[:pet_id])
+    favorite.add_pet(pet.id)
+    session[:favorite] = favorite.contents
+    flash[:ding_dong] = "#{pet.name} has been added to your favorites list ❤️"
+    redirect_to "/pets/#{params[:pet_id]}"
+  end
+
+  def index
+    @heading = "You have no favorite pets 💔"
+    @pets = []
+    if !favorite.contents.empty?
+      @heading = "My Favorites ❤️"
+      Pet.find_each do |pet|
+        pet_id_string = pet.id.to_s
+        @pets << pet if favorite.contents.include?(pet_id_string)
+      end
+    end
+  end
+
```

Comment:
Maybe this login in the index method could be refactored and moved to the favorite model?

Reply:
I thing thats a great idea. We should open a new branch for that fix